src/classes/UTIL_HtmlOutput_CTRL.cls

```diff
@@ -0,0 +1,55 @@
+public with sharing class UTIL_HtmlOutput_CTRL {
```

Comment:
@force2b @davisagli - Here is another idea, still fairly simple approach with the exception of the url's that we are changing to placeholders. It would take a bit of cleanup on the translation files and custom labels to swap everything for the placeholders.
A couple things I like about this approach, it's easier on translators to read the content and it provides one location for all of our page labels. I have only fixed the URL's on the home panel at this point. Just wanted to get early feedback before I take this too far. Thanks!

Reply:
@sam-knox - thoughts on this one? We link to docs in the custom labels which are translatable but have historically never been modified by translators. How do we deal with translating the docs themselves?

Reply:
I got side tracked and didn't commit my earlier comment and its now no longer relevant, so ...

I also like this in the apex controller, but I'm not really sure I like hardcoding target URL's into apex and relying on a Dev/Doc/Translator to know how to reference a link. Plus, adding a new link would require modifying the apex rather than just updating the label. I guess the question to be asked here is whether this is a workable solution to the root XSS issue we're trying to solve. If this class logic covers every possible vulnerability that could be implemented on a custom label, and we're confident that it could be documented clearly for everyone to know how to use it, then I think it could work. One thing I might suggest rather than using {0} and {1} - since it's possible that there might be some other code that wants to use these - could it instead use something like `{URL-START}` and `{URL-END}`? I think that might make the custom label easier to understand as well.

Also, I've seen `<p>`, `<ul>`, and `<li>` used in our tags as well. I wouldn't be surprised if there are others, such as `<hr>`, possibly in our underlying packages.

Reply:
Ok, third times the charm? I still have testing to do but wanted to get it back out here to get your feedback. So far this is my favorite approach, we whitelist the allowed URLs which allows them to change the page to the docs if needed. I know there are other places we are using different html tags but I thought we could add them as we need them since all of the settings labels only had bold and breaks so far.

Reply:
This approach seems okay to me. I haven't reviewed the PR closely because the VF autoformatting makes it hard to spot the consequential changes.

Reply:
I try to make the work commits separate from the auto formatting commits to help with that a bit, in case your curious about what actually changed. This [commit](https://github.com/SalesforceFoundation/Cumulus/pull/3858/commits/5183cff4f3155e7fb6832bca2ed5c34c676a6639) will show the new component in use.

Reply:
@force2b - How are you feeling about this approach? If you like the approach this PR is ready for Code Review. Thanks!

Reply:
I like where you're heading with this, but I do find it's a bit fragile. For example, the following two URL's should be effectively identical, but the result of passing through this controller breaks the one with the space:
* `<a href="https://powerofus.force.com/NPSP_Documentation">`
* `<a href="https://powerofus.force.com/NPSP_Documentation" >`

Do you think it's possible to strip spaces in some places, such as converting ` >` to just `>`?

Reply:
I changed this around a bit and allowed closing tags in general and whitelisted specific opening tags. This should prevent any issues with spaces on the back end of the url. I also added spacing and additional tags based on your feedback. Thanks!